core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala

```diff
@@ -0,0 +1,601 @@
+    consumerGroupCommand.close()
+  }
+
+  @Test
+  def testResetOffsetsToZonedDateTime() {
+    AdminUtils.createTopic(zkUtils, topic1, 1, 1)
+    TestUtils.produceMessages(servers, topic1, 50, acks = 1, 100 * 1000)
+
+    val format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX")
+    val checkpoint = new Date()
+
+    TestUtils.produceMessages(servers, topic1, 50, acks = 1, 100 * 1000)
+
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--reset-offsets", "--group", group, "--all-topics")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic1)
+
+    TestUtils.waitUntilTrue(() => {
+      val (_, assignmentsOption) = consumerGroupCommand.describeGroup()
+      assignmentsOption match {
+        case Some(assignments) =>
+          val sumOffset = (assignments.filter(_.topic.exists(_ == topic1))
+            .filter(_.offset.isDefined)
+            .map(assignment => assignment.offset.get) foldLeft 0.toLong)(_ + _)
+          sumOffset == 100
+        case _ => false
+      }
+    }, "Expected that consumer group has consumed all messages from topic/partition.")
+
+    executor.shutdown()
+
+    val cgcArgs1 = Array("--bootstrap-server", brokerList, "--reset-offsets", "--group", group, "--all-topics", "--to-datetime", format.format(checkpoint), "--execute")
+    val opts1 = new ConsumerGroupCommandOptions(cgcArgs1)
+    val consumerGroupCommand1 = new KafkaConsumerGroupService(opts1)
+
+    TestUtils.waitUntilTrue(() => {
+      val assignmentsToReset = consumerGroupCommand1.resetOffsets()
+      assignmentsToReset.exists { assignment => assignment._2.offset() == 50 }
+    }, "Expected the consumer group to reset to when offset was 50.")
+
+    printConsumerGroup()
+
+    AdminUtils.deleteTopic(zkUtils, topic1)
+    consumerGroupCommand.close()
+  }
+
+  @Test
+  def testResetOffsetsByDuration() {
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--reset-offsets", "--group", group, "--all-topics", "--by-duration", "PT1M", "--execute")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    AdminUtils.createTopic(zkUtils, topic1, 1, 1)
+
+    produceConsumeAndShutdown(consumerGroupCommand, 1, topic1, 100)
+
+    TestUtils.waitUntilTrue(() => {
+        val assignmentsToReset = consumerGroupCommand.resetOffsets()
+        assignmentsToReset.exists { assignment => assignment._2.offset() == 0 }
+    }, "Expected the consumer group to reset to offset 0 (earliest by duration).")
+
+    printConsumerGroup()
+
+    AdminUtils.deleteTopic(zkUtils, topic1)
+    consumerGroupCommand.close()
+  }
+
+  @Test
+  def testResetOffsetsByDurationToEarliest() {
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--reset-offsets", "--group", group, "--all-topics", "--by-duration", "PT0.1S", "--execute")
```

Comment:
nit: it's easier to read this when you put the arguments on separate lines like you did for some of the test cases.